flocker/node/agents/functional/test_ebs.py

```diff
@@ -182,6 +188,132 @@ def test_next_device_in_use(self):
                                        {u"/dev/sdf"})
         self.assertEqual(result, u"/dev/sdg")
 
+    def test_create_volume_gold_profile(self):
+        """
+        Requesting ``gold`` profile during volume creation honors
+        ``gold`` attributes.
+        """
+        self._assert_create_volume_with_mandatory_profile(
+            MandatoryProfiles.GOLD)
+
+    @capture_logging(assertHasMessage, CREATE_VOLUME_FAILURE)
+    def test_create_too_small_volume_gold_profile(self, logger):
+        """
+        Too small volume for ``gold`` attributes creates volume with
+        ``default`` profile.
+        """
+        self._assert_create_volume_with_mandatory_profile(
+            MandatoryProfiles.GOLD, created_profile=MandatoryProfiles.DEFAULT,
+            size_GiB=1)
+
+    def test_create_largest_volume_gold_profile(self):
+        """
+        Largest volume with ``gold`` profile.
+        """
+        raise SkipTest(u'This test is expensive (money).')
+        self._assert_create_volume_with_mandatory_profile(
+            MandatoryProfiles.GOLD, size_GiB=16*1024)
+
+    def test_create_too_large_volume_gold_profile(self):
```

Comment:
The aim was to try to create a volume so large that retry failed volume creation with downgraded profile does not help make progress. You are right in pointing out little value in these set of tests. Combined all three test_create_largest_volume_{gold,silver,bronze}_profile into one test. Please let me know what you think of the new version. Thanks!